pandas/tests/frame/test_missing.py

```diff
@@ -516,6 +516,22 @@ def test_fillna_skip_certain_blocks(self):
         # it works!
         df.fillna(np.nan)
 
+    @pytest.mark.parametrize("type", [int, float])
```

Comment:
I don't think these need to be parametrized - the dtype of the calling object is unrelated to the change here

Reply:
The dtype was exactly what was causing the issue though, this test failed before my change. It has to do with IntBlock and FloatBlock having _can_hold_na set to different values, causing a return in blocks.py's fillna method before it checked the validity of limit.

Reply:
Maybe np.random.uniform should be used here instead of np.random.rand which returns all zeros when cast to int, although there isn't much need for the change.